src/merkle/store/tests.rs

```diff
@@ -14,17 +14,39 @@ const LEAVES4: [Word; 4] = [
 ];
 
 #[test]
-fn test_root_not_in_store() -> Result<(), MerkleError> {
+fn test_building_tree() -> Result<(), MerkleError> {
+    let mut store = MerkleStore::default();
+
+    // starts with a empty tree of depth 2 (leafs are included in the constructor), and adds
+    // individual leafs until a full Merkle tree with depth 2 is built.
+    let root = *EmptySubtreeRoots::get_root_for_depth(2);
```

Comment:
I also want to double check the store is using the correct nodes, the order may be reversed.